x/wasm/internal/keeper/keeper.go

```diff
@@ -255,6 +255,23 @@ func (k Keeper) Migrate(ctx sdk.Context, contractAddress sdk.AccAddress, caller
 	return &value, nil
 }
 
+// UpdateContractAdmin sets the admin value on the ContractInfo. New admin can be nil to disable further migrations/ updates.
+func (k Keeper) UpdateContractAdmin(ctx sdk.Context, contractAddress sdk.AccAddress, caller sdk.AccAddress, newAdmin sdk.AccAddress) error {
+	contractInfo := k.GetContractInfo(ctx, contractAddress)
+	if contractInfo == nil {
+		return sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "unknown contract")
+	}
+	if contractInfo.Admin == nil {
+		return sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "migration not supported by this contract")
+	}
+	if !contractInfo.Admin.Equals(caller) {
+		return sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "no permission")
+	}
+	contractInfo.Admin = newAdmin
+	k.setContractInfo(ctx, contractAddress, *contractInfo)
```

Comment:
pp: `GetContactInfo` returns `*ContractInfo`, but `setContractInfo` accepts `ContractInfo`. Nice to have them both take pointers or both structs, not a mix.

But if there is a technical reason for it, no big deal, just seems like an odd mismatch

Reply:
🤔 `GetContactInfo` supports nil value as a result. I don't know if this is technically possible but refactoring this to a non pointer type would probably require an `error` return type as well. I will make them both pointers as a quick solution. I don't have a strong opinion. For me values are usually cleaner in api design as there are no modifications. But `setContractInfo` is a private method so does not matter much.